sdk/search/search-documents/swagger/Service.md

```diff
@@ -72,8 +72,44 @@ directive:
 ```yaml
 directive:
   - from: swagger-document
-    where: $.definitions.DataType
-    transform: >
+    where: $.definitions.SearchFieldDataType
+    transform: >
+      if ($["x-ms-enum"].values.length === 8) {
+        const newValues = $["x-ms-enum"].values.slice(0);
+        newValues.push({
+          "value": "Collection(Edm.String)",
+          "name": "Collection(Edm.String)"
+        });
+        newValues.push({
+          "value": "Collection(Edm.Int32)",
+          "name": "Collection(Edm.Int32)"
+        });
+        newValues.push({
+          "value": "Collection(Edm.Int64)",
+          "name": "Collection(Edm.Int64)"
+        });
+        newValues.push({
+          "value": "Collection(Edm.Double)",
+          "name": "Collection(Edm.Double)"
+        });
+        newValues.push({
+          "value": "Collection(Edm.Boolean)",
+          "name": "Collection(Edm.Boolean)"
+        });
+        newValues.push({
+          "value": "Collection(Edm.DateTimeOffset)",
+          "name": "Collection(Edm.DateTimeOffset)"
+        });
+        newValues.push({
+          "value": "Collection(Edm.GeographyPoint)",
+          "name": "Collection(Edm.GeographyPoint)"
+        });
+        newValues.push({
+          "value": "Collection(Edm.ComplexType)",
+          "name": "Collection(Edm.ComplexType)"
+        });
+        $["x-ms-enum"].values = newValues;
+      }
       if ($.enum.length === 8) {
```

Comment:
the code below here seems like it will never execute now and can be removed?

It does seem a bit of a shame that we replaced a loop with a hardcoded list of values

Reply:
Agreed that hardocoding looks bad. I really tried different methods. But could not get it working. 

Also, we need both conditions here. The final output will look like:


```json
"SearchFieldDataType": {
      "type": "string",
      "enum": [
        "Edm.String",
        "Edm.Int32",
        "Edm.Int64",
        "Edm.Double",
        "Edm.Boolean",
        "Edm.DateTimeOffset",
        "Edm.GeographyPoint",
        "Edm.ComplexType",
        "Collection(Edm.String)",
        "Collection(Edm.Int32)",
        "Collection(Edm.Int64)",
        "Collection(Edm.Double)",
        "Collection(Edm.Boolean)",
        "Collection(Edm.DateTimeOffset)",
        "Collection(Edm.GeographyPoint)",
        "Collection(Edm.ComplexType)"
      ],
      "x-ms-enum": {
        "name": "SearchFieldDataType",
        "modelAsString": true,
        "values": [
          {
            "value": "Edm.String",
            "name": "String",
            "description": "Indicates that a field contains a string."
          },
          {
            "value": "Edm.Int32",
            "name": "Int32",
            "description": "Indicates that a field contains a 32-bit signed integer."
          },
          {
            "value": "Edm.Int64",
            "name": "Int64",
            "description": "Indicates that a field contains a 64-bit signed integer."
          },
          {
            "value": "Edm.Double",
            "name": "Double",
            "description": "Indicates that a field contains an IEEE double-precision floating point number."
          },
          {
            "value": "Edm.Boolean",
            "name": "Boolean",
            "description": "Indicates that a field contains a Boolean value (true or false)."
          },
          {
            "value": "Edm.DateTimeOffset",
            "name": "DateTimeOffset",
            "description": "Indicates that a field contains a date/time value, including timezone information."
          },
          {
            "value": "Edm.GeographyPoint",
            "name": "GeographyPoint",
            "description": "Indicates that a field contains a geo-location in terms of longitude and latitude."
          },
          {
            "value": "Edm.ComplexType",
            "name": "Complex",
            "description": "Indicates that a field contains one or more complex objects that in turn have sub-fields of other types."
          },
          {
            "value": "Collection(Edm.String)",
            "name": "Collection(Edm.String)"
          },
          {
            "value": "Collection(Edm.Int32)",
            "name": "Collection(Edm.Int32)"
          },
          {
            "value": "Collection(Edm.Int64)",
            "name": "Collection(Edm.Int64)"
          },
          {
            "value": "Collection(Edm.Double)",
            "name": "Collection(Edm.Double)"
          },
          {
            "value": "Collection(Edm.Boolean)",
            "name": "Collection(Edm.Boolean)"
          },
          {
            "value": "Collection(Edm.DateTimeOffset)",
            "name": "Collection(Edm.DateTimeOffset)"
          },
          {
            "value": "Collection(Edm.GeographyPoint)",
            "name": "Collection(Edm.GeographyPoint)"
          },
          {
            "value": "Collection(Edm.ComplexType)",
            "name": "Collection(Edm.ComplexType)"
          }
        ]
      },
      "description": "Defines the data type of a field in a search index."
    }
```

Reply:
Oh I understand now. I originally mistakenly thought you simply had to update "x-ms-enum" values instead.

I'm guessing you tried something like this already:

```
 if ($["x-ms-enum"].values.length === 8) {
    const newValues = $["x-ms-enum"].values.slice(0);
    for (let value of $["x-ms-enum"].values) {
       const name = 'Collection('+value+')';
       newValues.push({"value": name, "name": name});
    }
    $["x-ms-enum"].values = newValues;
}
```

Reply:
Yeap. Autorest trasnformations are complaining. So, switched to hardcoding :-)